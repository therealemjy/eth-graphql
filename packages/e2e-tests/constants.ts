export const MAINNET_CHAIN_ID = 1;
