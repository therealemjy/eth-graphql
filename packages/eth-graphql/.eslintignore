coverage
.turbo
dist
node_modules
