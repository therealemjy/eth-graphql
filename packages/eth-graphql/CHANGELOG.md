# eth-graphql

## 1.0.0

### Major Changes

- 500593f: Change GraphiQL route

## 0.3.10

### Patch Changes

- Update imports to improve compatibility with projects

## 0.3.9

### Patch Changes

- Import ApolloLink from @apollo/client

## 0.3.8

### Patch Changes

- Revert back to 0.3.6

## 0.3.7

### Patch Changes

- bc9bfb0: Update how config is loaded to handle TS path aliases

## 0.3.6

### Patch Changes

- Fix links in README

## 0.3.5

### Patch Changes

- ec05d44: Update README

## 0.3.4

### Patch Changes

- Update documentation to add enums

## 0.3.3

### Patch Changes

- Update README

## 0.3.2

### Patch Changes

- Remove unnecessary package

## 0.3.1

### Patch Changes

- Fix issue with calls not getting aggregated

## 0.3.0

### Minor Changes

- Fix README

## 0.2.0

### Minor Changes

- Add README

## 0.1.0

### Minor Changes

- First release

## 0.0.1

### Patch Changes

- Initialize project
