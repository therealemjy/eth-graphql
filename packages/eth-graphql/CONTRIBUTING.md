# Contributing to eth-graphql

First off, thanks for taking the time to contribute!

All types of contributions are encouraged and valued. However, please take the time to read this
document and search for any existing pull requests or issues that might already touch on what you
wish to add or improve. Also, before making any pull requests, please open an issue describing what
feature you wish to add to the library or what bug you're trying to fix. Unsolicited pull requests
will very likely be closed as I'm trying to keep things organized.

If you like the project, but just don't have time to contribute, that's fine. There are other easy
ways to support the project and show your appreciation, which I would also be very happy about:

- Star the repository
- Tweet about it
- Use this library in your projects
- Mention eth-graphql at local meetups and tell your friends/colleagues

## Setting up your development environment

### Install dependencies

The project uses yarn to manage dependencies:

```bash
yarn install
```

### Start your development environment

```bash
yarn dev
```

This command will build the eth-graphql package and spin up the [example](../../example/) app. Open
[http://localhost:3000](http://localhost:3000) with your browser to see the result.

You can use the example app to test any changes you made to eth-graphql, however, make sure to
revert those changes before you submit a pull request (unless of course the purpose of your PR is to
bring changes to the example app).

### Testing

The app is tested through unit and end-to-end tests. The unit tests cover utility functions and
small modules. Bigger modules are tested through end-to-end tests.

To run all the tests on the project, you can use the command:

```bash
yarn test
```

If you wish to run the end-to-end tests only, use the command:

```bash
yarn test:e2e
```

To run the unit tests only, use:

```bash
yarn test:units
```

### Useful scripts

Linting:

```bash
yarn lint
```

Typecheck:

```bash
yarn tsc
```

Formatting:

```bash
yarn format
```

### Submitting a pull request

Once you have finished working on your feature or bug fix and wish to open a pull request, you will
first need to run the following command:

```bash
yarn changeset
```

You will be guided through a series of questions in order to describe the changes you're making and
identify how they should update the package version. As a reference:

- Breaking changes should trigger a major bump (e.g.: you've updated the config type and that change
  will require existing users to update their setup)
- Non-breaking changes that add new functionalities should trigger a minor bump (e.g.: you've added
  a new field to the config, without altering the rest of the config nor the existing
  functionalities)
- Non-breaking changes that do not add any functionality should trigger a patch bump (e.g.: you've
  fixed a bug without altering any of the existing functionalities)

Make sure to commit the changes brought to the repository after running this command. You are then
ready to submit your pull request against the `main` branch. I will do my best to review pull
requests in a timely manner; please be mindful of my time since I'm not a full time open source
contributor.
