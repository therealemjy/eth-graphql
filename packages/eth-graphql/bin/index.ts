#!/usr/bin/env node

console.log("HEY HEY");
