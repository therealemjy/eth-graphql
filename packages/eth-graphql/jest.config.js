module.exports = {
  preset: 'ts-jest',
  testEnvironment: 'node',
  resetMocks: true,
  collectCoverageFrom: ['src/**/*.{ts,js}'],
  modulePathIgnorePatterns: ['./src/cli/__tests__/fakeConfig'],
};
