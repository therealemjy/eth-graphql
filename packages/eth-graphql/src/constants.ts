export const PORT = 8008;
export const GRAPHIQL_ROUTE = '/eth-call-graphiql';
