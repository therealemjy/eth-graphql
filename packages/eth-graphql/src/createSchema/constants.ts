export const ROOT_FIELD_NAME = 'contracts';
export const ROOT_FIELD_SINGLE_ARGUMENT_NAME = 'addresses';
export const MULT_FIELD_SUFFIX = '_MULT';
export const MULT_FIELD_SINGLE_ARGUMENT_NAME = 'args';
