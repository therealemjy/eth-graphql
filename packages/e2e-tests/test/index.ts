import { ApolloClient, InMemoryCache, gql } from '@apollo/client';
import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
import { expect } from 'chai';
import chai from 'chai';
import chaiJestSnapshot from 'chai-jest-snapshot';
import { createLink } from 'eth-graphql';
import { ethers } from 'hardhat';

import { MAINNET_CHAIN_ID } from '../constants';

chai.use(chaiJestSnapshot);

before(function () {
  chaiJestSnapshot.resetSnapshotRegistry();
});

beforeEach(function () {
  chaiJestSnapshot.configureUsingMochaContext(this);
});

// Create fixture for TestContract
async function deployTestContractFixture() {
  const TestContract = await ethers.getContractFactory('TestContract');
  const testContract = await TestContract.deploy();

  const testContractAddress = testContract.address;
  console.log('testContractAddress', testContractAddress);

  // TODO: Mock config to include deployed contracts address

  const link = createLink({
    // @ts-ignore
    provider: ethers.provider,
  });

  const client = new ApolloClient({
    cache: new InMemoryCache(),
    link,
  });

  return client;
}

describe('end-to-end tests', function () {
  it('should return the correct data', async function () {
    const client = await loadFixture(deployTestContractFixture);

    // Make GraphQL request
    const { data } = await client.query({
      query: gql`
        query ($chainId: Int!) {
          contracts(chainId: $chainId) {
            TestContract {
              getMovieTitle
            }
          }
        }
      `,
      variables: {
        chainId: MAINNET_CHAIN_ID,
      },
    });

    console.log(data);

    expect(data).to.matchSnapshot();
  });
});
