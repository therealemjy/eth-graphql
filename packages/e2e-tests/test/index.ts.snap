// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end tests should return the correct data 1`] = `
Object {
  "contracts": Object {
    "TestContract": Object {
      "__typename": "TestContract",
      "getAddress": "0xA5ae0b2386De51Aba852551A1EE828BfD598E111",
      "getAllMovies": Array [
        Object {
          "__typename": "TestContract_Movie",
          "director": Object {
            "__typename": "TestContract_Director",
            "name": "some director",
            "walletAddress": "0xA5ae0b2386De51Aba852551A1EE828BfD598E111",
          },
          "id": "0",
          "status": "1",
          "title": "some movie",
        },
        Object {
          "__typename": "TestContract_Movie",
          "director": Object {
            "__typename": "TestContract_Director",
            "name": "Robert Zemeckis",
            "walletAddress": "0x86f11f319E53481493Ea50d97eAc684c5Ca8403D",
          },
          "id": "1",
          "status": "1",
          "title": "Forrest Gump",
        },
      ],
      "getAnyMovie": Object {
        "__typename": "TestContract_Movie",
        "id": "0",
        "status": "1",
        "title": "some movie",
      },
      "getBoolean": true,
      "getBytes": "0xed6c11b0b5b808960df26f5bfc471d04c1995b0ffd2055925ad1be28d6baadfd",
      "getInt": "125637412567341",
      "getMultipleValues": Object {
        "__typename": "getMultipleValuesOutput",
        "value0": "Catch Me If You Can",
        "value1": "0",
        "value2": Object {
          "__typename": "TestContract_Director",
          "name": "Steven Spielberg",
          "walletAddress": "0xA5ae0b2386De51Aba852551A1EE828BfD598E111",
        },
      },
      "getNamedString": "Catch Me If You Can",
      "getString": "string example",
      "getTuple": Array [
        "0",
        "1",
        "2",
      ],
      "getUint": "1256374125673412563451263546712536712536712573612",
      "mayhem": Object {
        "__typename": "mayhemOutput",
        "director": Object {
          "__typename": "TestContract_Director",
          "name": "some director",
          "walletAddress": "0xA5ae0b2386De51Aba852551A1EE828BfD598E111",
        },
        "passedMovies": Array [
          Object {
            "__typename": "TestContract_Movie",
            "director": Object {
              "__typename": "TestContract_Director",
              "name": "some director",
              "walletAddress": "0xA5ae0b2386De51Aba852551A1EE828BfD598E111",
            },
            "id": "0",
            "status": "1",
            "title": "some movie",
          },
          Object {
            "__typename": "TestContract_Movie",
            "director": Object {
              "__typename": "TestContract_Director",
              "name": "some director",
              "walletAddress": "0xA5ae0b2386De51Aba852551A1EE828BfD598E111",
            },
            "id": "0",
            "status": "1",
            "title": "some movie",
          },
        ],
        "statuses": Array [
          "0",
          "1",
          "1",
          "1",
        ],
        "value1": "671432189",
      },
      "movies": Object {
        "__typename": "moviesOutput",
        "director": Object {
          "__typename": "TestContract_Director",
          "name": "Steven Spielberg",
          "walletAddress": "0xA5ae0b2386De51Aba852551A1EE828BfD598E111",
        },
        "id": "0",
        "status": "0",
        "title": "Catch Me If You Can",
      },
      "overloadedFn0": Object {
        "__typename": "overloadedFnOutput",
        "value0": "overloaded result 1",
        "value1": "1",
      },
      "overloadedFn1": Object {
        "__typename": "overloadedFnOutput",
        "value0": "overloaded result 0",
        "value1": "0",
      },
      "overloadedFn2": Object {
        "__typename": "overloadedFnOutput",
        "value0": "overloaded result 2",
        "value1": "2",
      },
      "passAddress": "0xA5ae0b2386De51Aba852551A1EE828BfD598E111",
      "passBoolean": true,
      "passBytes": "0xed6c11b0b5b808960df26f5bfc471d04c1995b0ffd2055925ad1be28d6baadfd",
      "passInt": "1265341",
      "passMovie": Object {
        "__typename": "TestContract_Movie",
        "director": Object {
          "__typename": "TestContract_Director",
          "name": "some director",
          "walletAddress": "0xA5ae0b2386De51Aba852551A1EE828BfD598E111",
        },
        "id": "0",
        "status": "1",
        "title": "some movie",
      },
      "passString": "some-string",
      "passTuple": Array [
        "0",
        "1",
        "2",
      ],
      "passUint": "128738121231267831231323",
      "passUnnamedString": "passUnnamedString result",
    },
    "__typename": "contracts",
  },
}
`;
